### PR TITLE
amend to onRouterNavigate

### LINKED ISSUE #21: pageIncompletePrompt can break the assessmentResults 'retry' functionality
### Subject of the issue/enhancement/features
Clicking 'retry' on the assessment results component can cause the pageIncompletePrompt to activate in some setups. Clicking 'Yes' in the PIP prompt closes the prompt but the assessment does not reset.

### Steps to reproduce
Either amend adapt-contrib-assessmentResults.js so that it only marks as complete if the user passes the assessment or add some additional content below the assessment results component. Then take the assessment, fail it and click the 'Retry Assessment' button in the assessmentResults component.

### Expected behaviour
No PIP prompt should be shown and the assessment should reset.

### Actual behaviour
A PIP prompt is displayed and the following error is displayed in the console:
`Adapt.findById() unable to find collection type for id: undefined`
Click 'Yes' in the PIP prompt closes the prompt but the assessment does not reset. Clicking the 'Retry Assessment' button again at this point does then cause it to reset as expected.

### PR DESCRIPTION
**Pull request by moloko**
to fix #21 

also re-write `allComponentsComplete` so it's much shorter

### CODE REVIEW

File: js/adapt-pageIncompletePrompt.js [hunk 1]
```diff
@@ -79,27 +78,31 @@ define([
...
-            
+
             if(!this.isEnabled() || this.allComponentsComplete()) return;
 
             this.href = window.location.href;
 
             if (routeArguments[0]) {
-                //check if routing to current page child
-                //exit if on same page
+                var id = routeArguments[0];
```
Comment (tomgreenfield) [inline js/adapt-pageIncompletePrompt.js]: Move `id` higher up and make conditional `if (id) {`?

File: js/adapt-pageIncompletePrompt.js [hunk 2]
```diff
@@ -172,18 +175,11 @@ define([
...
-                var isMandatory = (component.get('_isOptional') === false);
-                var isComplete = component.get("_isComplete");
-            
-                if(isMandatory && !isComplete) return false;
-            }
-            
-            return true;
+
+            return this.pageComponents.every(function(component) {
+                if(component.get('_isComplete') || component.get('_isOptional')) return true;
```
Comment (tomgreenfield) [inline js/adapt-pageIncompletePrompt.js]: Can reduce to the below?
```js
return component.get('_isComplete') || component.get('_isOptional');
```

File: js/adapt-pageIncompletePrompt.js [hunk 3]
```diff
@@ -79,27 +78,31 @@ define([
...
+                // exit if on same page (e.g. if doing 'retry assessment')
+                if (id === Adapt.location._currentId) return;
+                // check if routing to current page child
                 try {
-                    var id = routeArguments[0];
                     var model = Adapt.findById(id);
                     var parent = model.findAncestor("contentObjects");
                     if (parent.get("_id") == this.pageModel.get("_id")) return;
-                } catch (e) {}
+                } catch (e) {
```
Comment (tomgreenfield) [inline js/adapt-pageIncompletePrompt.js]: Is the `try...catch` still required?

Comment (moloko) [reply js/adapt-pageIncompletePrompt.js]: I'm not sure TBH, I'll defer to @oliverfoster on that one...

Comment (moloko) [reply js/adapt-pageIncompletePrompt.js]: it does catch other errors that haven't been specifically 'thrown' though - for example the issue I'm fixing with this PR was previously being caught and handled by the `catch` block - which is why I added the `return` to it as I don't think the prompt should be shown if an error was encountered at this point.

**Pull request merged**